core/genesis.go

```diff
@@ -229,8 +229,8 @@ func (g *Genesis) trieConfig() *triedb.Config {
 }
 
 // TODO: migrate this function to "flush" for more similarity with upstream.
-func (g *Genesis) toBlock(db ethdb.Database, triedb *triedb.Database) *types.Block {
-	statedb, err := state.New(types.EmptyRootHash, state.NewDatabaseWithNodeDB(db, triedb), nil)
```

Comment:
I feel like this PR has a little context and the intention is not clear. Commented here https://github.com/ava-labs/coreth/pull/959/files#r2114807070